include/kimera_vio_ros/RosDataProviderInterface.hpp

```diff
@@ -48,12 +65,14 @@ struct PointNormalUV {
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 } EIGEN_ALIGN16;
 
-class RosDataProviderInterface : public DataProviderInterface {
+class RosDataProviderInterface : public DataProviderInterface, public rclcpp::Node{
  public:
   KIMERA_DELETE_COPY_CONSTRUCTORS(RosDataProviderInterface);
   KIMERA_POINTER_TYPEDEFS(RosDataProviderInterface);
 
-  RosDataProviderInterface();
+  RosDataProviderInterface(
+      const std::string & node_name,
+      const rclcpp::NodeOptions & options = rclcpp::NodeOptions());
```

Comment:
It would be nice if you use our clang code style so that we are all on the same page:
We just use google cpp code style: https://github.mit.edu/SPARK/Kimera-VIO/blob/master/.clang-format

Here the issue is that the `&` should be next to the variable `const std::string&`. Minor for now, but it adds up quickly.

Reply:
Woops, I've been switching to rust as I never could remember all of C++ code styles. I was going to enable official ROS2 linters before refactoring for switching the PR from WIP. It also is based on Google Style guide and then some.

https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#id1
https://answers.ros.org/question/306194/ros1-and-ros2-linting/?answer=306199#post-id-306199